File: 1uzd/1uzd.cpp
//Vaidotas Gruzas PI17A
#include <iostream>
using namespace std;
int main()
{
    int x, X=999;//kintamieji
    cout<<"Iveskite dvimacio masyvo eiluciu skaiciu: "<<endl;
    cin>>x;
    int A[x][x];
    char B[x][x];//konvertuotas masyvas
    cout<<"Is eiles, nuo virsutinio desiniojo kampo ivardinkite dvimacio masyvo elementus: "<<endl;
    for(int i=0; i<x; i++)
    {
        for(int j=0; j<x; j++)
        {
            cin>>A[i][j];
        }
    }
    for(int i=0; i<x; i++)
    {
        if(X>A[i][i]) X=A[i][i];
    }
    cout<<"Konvertuotas masyvas."<<endl;
    for(int i=0; i<x; i++)
    {
        for(int j=0; j<x; j++)
        {
            if(A[i][j]>X) B[i][j]='D';
                else if(A[i][j]<X) B[i][j]='M';
                    else B[i][j]='L';
            cout<<B[i][j]<<" ";
        }
        cout<<endl;
    }
    return 0;
}

File: 1uzd/1uzd.rb
puts 'Iveskite dvimacio masyvo eiluciu skaiciu:'
size = gets.chomp.to_i
array = Array.new(size) { Array.new(size) }
array_min = Array.new(size)
puts 'Is eiles, nuo virsutinio desiniojo kampo ivardinkite dvimacio masyvo
elementus.'
size.times do |i|
  size.times do |j|
    array[i][j] = gets.chomp.to_i
    array_min[i] = array[i][i]
  end
end
puts 'Konvertuotas masyvas:'
size.times do |i|
  size.times do |j|
    array[i][j] = if array[i][j] > array_min.min
                    'D'
                  elsif array[i][j] < array_min.min
                    'M'
                  else
                    'L'
                  end
  end
end
puts(array) & :join
